api/v1alpha1/api.go

```diff
@@ -1,3 +1,118 @@
+//
+// The InputSchema is used to determine the structure of the requests that the Gateway will
+// receive. And then the Gateway will route the traffic to the appropriate LLMBackend based
+// on the output schema of the LLMBackend while doing the other necessary jobs like rate limit, etc.
+type LLMRoute struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines the details of the LLM policy.
+	Spec LLMRouteSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// LLMRouteList contains a list of LLMTrafficPolicy
+type LLMRouteList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []LLMRoute `json:"items"`
+}
+
+// LLMRouteSpec details the LLMRoute configuration.
+type LLMRouteSpec struct {
+	// APISchema specifies the API schema of the input that the target Gateway(s) will receive.
+	// Based on this schema, the ai-gateway will perform the necessary transformation to the
+	// output schema specified in the selected LLMBackend during the routing process.
+	APISchema LLMAPISchema `json:"inputSchema"`
+	// TargetRefs are the names of the Gateway resources this policy is being attached to.
+	// The namespace is "local", i.e. the same namespace as the LLMRoute.
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=128
+	TargetRefs []gwapiv1a2.LocalPolicyTargetReferenceWithSectionName `json:"targetRefs"`
+	// BackendRefs lists the LLMBackends that this LLMRoute will route traffic to.
+	// The namespace is "local", i.e. the same namespace as the LLMRoute.
+	//
+	// +kubebuilder:validation:MaxItems=128
+	BackendRefs []LLMBackendLocalRef `json:"backendRefs,omitempty"`
+}
+
+// LLMBackendLocalRef is a reference to a LLMBackend resource in the "local" namespace.
+type LLMBackendLocalRef struct {
+	// Name is the name of the LLMBackend in the same namespace as the LLMRoute.
+	Name string `json:"name"`
+}
+
 // +kubebuilder:object:root=true
+
+// LLMBackend is a resource that represents a single backend for LLMRoute.
+// A backend is a service that handles traffic with a concrete API specification.
+type LLMBackend struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines the details of the LLM policy.
+	Spec LLMBackendSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// LLMBackendList contains a list of LLMBackends.
+type LLMBackendList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []LLMBackend `json:"items"`
+}
+
+// LLMBackendSpec details the LLMBackend configuration.
+type LLMBackendSpec struct {
+	// APISchema specifies the API schema of the output that this LLMBackend can accept.
+	// Based on this schema, the ai-gateway will perform the necessary transformation for
+	// the pair of LLMRoute.APISchema and LLMBackend.APISchema.
```

Comment:
I am not sure if we want to support arbitrary transformation between any two of the APISchemas

Reply:
https://github.com/envoyproxy/ai-gateway/pull/20/commits/865ec35246232ff7ef24bffa0411798bba50fd91 done, thanks for the comment!